### PR TITLE
🔧 fix: fail on embedding dimension mismatch and redact DSNs in logs

### PR DESCRIPTION
**Pull request by yeazelm**
🧍When I was looking at bumping to new embedding configurations I noticed that the logs had the full DSN in them and the dimensions weren't properly checked so things could fail in unexpected ways when trying to move to the new embedding from an older version. 🧍

**Silent dimension mismatch.** `ensureSchema` uses `CREATE TABLE IF NOT EXISTS`, which keeps an existing table's declared vector dimension. A driver configured for 1024 dims connected to a `vector(1536)` table starts cleanly and logs `dimensions=1024` — then every embedding insert fails. pgvector can't resize columns in place, so the driver now checks the existing column's dimension at startup and fails with the options spelled out (re-embed into a new table, or drop the old one). Dimensions are also bounded to pgvector's 16000 vector-type max at construction.

**Database password in logs.** The "vector storage/search enabled" startup lines logged the raw DSN — tenant DB passwords are sitting in staging pod logs and downstream log stores. A new `config.RedactDSN` masks passwords in URL and keyword/value DSN forms, with a pattern fallback for URL-shaped strings that fail to parse so malformed DSNs can't leak through. `cli/sqlite-import`'s local `redactDSN` was rebuilt on the shared helper — it mutated `pgxpool.Config.ConnConfig.Password` and returned `ConnString()`, which returns the original unmodified string, so it never actually masked anything.

## Test plan

- [x] `dagger call test` — includes new pgvector integration specs (mismatched reopen fails with the new error; matching reopen succeeds) and `RedactDSN` unit specs for both DSN forms and edge cases
- [x] `make test-kafka-e2e`
- [ ] After staging rollout: pod startup log shows `tapes:xxxxx@` in `vector_store_target`

Part of PCC-654.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by jasonwc — APPROVED** [review]
LGTM, I noticed that DSN too but you beat me to the fix!

File: pkg/config/redact.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+// can be logged safely. Both URL DSNs (postgres://user:pass@host/db) and
+// keyword/value DSNs (host=... password=...) are supported. Strings without a
+// password are returned unchanged.
+func RedactDSN(dsn string) string {
+	if u, err := url.Parse(dsn); err == nil && u.Scheme != "" && u.User != nil {
+		if _, ok := u.User.Password(); ok {
+			u.User = url.UserPassword(u.User.Username(), redactedPassword)
+		}
+		return u.String()
+	}
```
[the remark this answers is not in the document]
Comment (yeazelm) [reply pkg/config/redact.go]: Fixed in the amended commit: `RedactDSN` now masks `?password=...` query parameters on parsed URLs (no longer gated on userinfo being present), and the unparseable-URL fallback chains both the userinfo and keyword patterns — with the keyword pattern stopping at `&` so adjacent query params survive. Specs added for query-param-only, both-locations, and unparseable-with-query cases.
